FWCore/ParameterSet/src/types.cc

```diff
@@ -459,9 +458,9 @@ bool edm::decode(double& to, std::string const& from) {
   else {
     try {
       // std::cerr << "from:" << from << std::endl;
-      to = boost::lexical_cast<double>(from);
+      to = std::stod(from);
       // std::cerr << "to:" << to << std::endl;
-    } catch (boost::bad_lexical_cast&) {
+    } catch (const std::invalid_argument&) {
```

Comment:
I think it would be useful to catch also `std::out_of_range`. The both appear to inherit from `std::logic_error`, so I'd suggest to catch either that or the `std::exception`.

Reply:
Adding a catch `std::out_of_range` in both the files.